packages/expo-updates/src/UseUpdatesUtils.ts

```diff
@@ -0,0 +1,90 @@
+import * as Updates from './Updates';
+import type { Manifest, UpdatesNativeStateMachineContext } from './Updates.types';
+import { UpdateInfoType, type CurrentlyRunningInfo, type UpdateInfo } from './UseUpdates.types';
+
+// The currently running info, constructed from Updates constants
+export const currentlyRunning: CurrentlyRunningInfo = {
+  updateId: Updates.updateId ?? undefined,
+  channel: Updates.channel ?? undefined,
+  createdAt: Updates.createdAt ?? undefined,
+  isEmbeddedLaunch: Updates.isEmbeddedLaunch,
+  isEmergencyLaunch: Updates.isEmergencyLaunch,
+  manifest: Updates.manifest ?? undefined,
+  runtimeVersion: Updates.runtimeVersion ?? undefined,
+};
+
+// Type for the state managed by useUpdates().
+// Used internally by this module and not exported publicly.
+export type UseUpdatesStateType = {
+  availableUpdate?: UpdateInfo;
+  downloadedUpdate?: UpdateInfo;
+  checkError?: Error;
+  downloadError?: Error;
+  isUpdateAvailable: boolean;
+  isUpdatePending: boolean;
+  isChecking: boolean;
+  isDownloading: boolean;
+  lastCheckForUpdateTimeSinceRestart?: Date;
+};
+
+// Constructs an UpdateInfo from a manifest
+export const updateFromManifest = (manifest?: Manifest) => {
+  return manifest
+    ? {
+        type: UpdateInfoType.NEW,
+        updateId: manifest?.id ?? undefined,
+        createdAt:
+          manifest && 'createdAt' in manifest && manifest.createdAt
+            ? new Date(manifest.createdAt)
+            : manifest && 'publishedTime' in manifest && manifest.publishedTime
+            ? new Date(manifest.publishedTime)
+            : // We should never reach this if the manifest is valid and has a commit time,
+              // but leave this in so that createdAt is always defined
+              new Date(0),
+        manifest: manifest ?? undefined,
+      }
+    : undefined;
+};
+
+// Constructs the availableUpdate from the native state change event context
+export const availableUpdateFromContext = (context: UpdatesNativeStateMachineContext) =>
+  updateFromManifest(context?.latestManifest);
+
+// Constructs the downloadedUpdate from the native state change event context
+export const downloadedUpdateFromContext = (context: UpdatesNativeStateMachineContext) =>
+  updateFromManifest(context?.downloadedManifest);
+
+// Default useUpdates() state
+export const defaultUseUpdatesState: UseUpdatesStateType = {
+  isChecking: false,
+  isDownloading: false,
+  isUpdateAvailable: false,
+  isUpdatePending: false,
+};
+
+// Transform the useUpdates() state based on native state machine context
+export const reduceUpdatesStateFromContext = (
+  updatesState: UseUpdatesStateType,
+  context: UpdatesNativeStateMachineContext
+) => {
+  if (context.isChecking) {
```

Comment:
I think we still want to convert the manifests from context and transfer the other fields from context into the state in this case (we don't know for sure they changes to them weren't included in this emit). Put another way, we still want all the code below to run, just also to set the `lastCheckForUpdateTimeSinceRestart` only in this case.

Reply:
Good catch, will do.